src/xFactor/Person.java

```diff
@@ -0,0 +1,76 @@
+package xFactor;
+
+public class Person {
+	private String firstName;
+	private String lastName;
+	private int age;
+	private String city;
+	private Gender value;
+	enum Gender {
+		Male, Female
+	}
+	
+	public Person(String fName, String lName, int age, Gender gender, String city) {
+		this.firstName = fName;
+		this.lastName = lName;
+		this.age = age;
+		this.value = gender;
+		this.city = city;
+	}
+	
+	public String getFirstName() {
+		return firstName;
+	}
+
+	public void setFirstName(String firstName) {
+		this.firstName = firstName;
+	}
+
+	public String getLastName() {
+		return lastName;
+	}
+
+	public void setLastName(String lastName) {
+		this.lastName = lastName;
+	}
+	
+	public String getName() {
+		return firstName + " " + lastName;
+	}
+	
+	public void setName(String fName, String lName) {
+		this.firstName = fName;
+		this.lastName = lName;
+	}
+	
+	public int getAge() {
+		return age;
+	}
+	
+	public void setAge(int age) {
+		this.age = age;
+	}
+
+	public String getCity() {
+		return city;
+	}
+	
+	public void setCity(String city) {
+		this.city = city;
+	}
+	
+	public void setValue(Gender e)
+    {
+        this.value = e; 
+    }
+
+    public Gender getValue()
+    {
+        return value;
+    }
+
+	@Override
+	public String toString() {
+		return this.getName();

```

Comment:
better for toString to print everything and you can call getName() in the print methods

Reply:
So every time a collection is printed, I should iterate over its elements and call getName()?